.github/workflows/test-install.yml

```
@@ -158,6 +158,7 @@ jobs:
 
     - name: Install aiida-core
       run: |
+        pip install numpy==1.19.4
```

I would suggest to make this step explicitly conditional and add a comment as to why it is necessary:
```suggestion
    # Work-around issue caused by pymatgen's setup process, which will install the latest 
    # numpy version (including release candidates) regardless of our actual specification:
    - name: Install numpy [py36]
      if: matrix.python-version == 3.6
      run: |
        pip install `grep 'numpy==' requirements/requirements-py-${{ matrix.python-version }}.txt`

    - name: Install aiida-core
      run: |
```

---

I see the idea of wanting to limit the scope to the bare minimum but I don't think this is the right thing to do in this case? With this scoped limitation, for all other versions, the release candidate of `numpy` will still get installed with all the potential breaking of code that comes with it. Maybe the install will be fine since the Python version is still compatible but other backwards incompatible changes or bugs may be introduced.

---

I indeed overlooked this, good point.

---

Ok, I will then merge this. Thanks for the review